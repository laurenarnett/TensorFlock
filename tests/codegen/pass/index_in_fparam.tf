main = v[i] * mat[i,j];

v : T<n>; // i : [0,n)
v[i] = cast (i + n); // evaluates to [5. 6. 7. 8. 9.] (if n = 5)

mat : T<n,m>;
mat[i,j] = cast (i * j);

deduce_n : T<n>;
deduce_n = [1. 2. 3. 4. 5.];

deduce_m : T<m>;
deduce_m = [1. 2. 3.];
