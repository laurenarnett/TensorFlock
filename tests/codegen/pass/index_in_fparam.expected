[5.00 6.00 7.00 8.00 9.00]
